### PR TITLE
add ng-add to store-devtools re #920

### PR DESCRIPTION
**Pull request by martzcodes**
Adds ng add for store-devtools and a maxAge option (defaults max age to 25 based on docs and comment https://github.com/ngrx/platform/issues/920#issuecomment-392279675

There were also 2 files that got changed via prettier

Related to #920

### CODE REVIEW

**Comment by martzcodes** [comment]
@brandonroberts Is this a CircleCI fail?  If so, can you restart the bazel task?

It works locally for me:

```
mattmartz@Matts-iMac  ~/Development/platform   store-devtools-add  bazel query --output=label //... | xargs bazel test
WARNING: Switched off --watchfs again... temporarily falling back to manually checking files for changes
INFO: Analysed 61 targets (1 packages loaded).
INFO: Found 56 targets and 5 test targets...
INFO: Elapsed time: 14.739s, Critical Path: 11.91s
INFO: 22 processes: 9 darwin-sandbox, 13 local.
INFO: Build completed successfully, 25 total actions
//modules/effects/spec:test                                              PASSED in 2.3s
//modules/entity/spec:test                                               PASSED in 1.6s
//modules/router-store/spec:test                                         PASSED in 1.9s
//modules/store-devtools/spec:test                                       PASSED in 1.9s
//modules/store/spec:test                                                PASSED in 1.8s

Executed 5 out of 5 tests: 5 tests pass.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see INFO: Build completed successfully, 25 total actions
```

File: example-app/app/books/components/book-detail.component.ts [hunk 1]
```diff
@@ -95,8 +95,8 @@ export class BookDetailComponent {
   get thumbnail() {
```
Comment (brandonroberts) [inline example-app/app/books/components/book-detail.component.ts]: Put the whitespace changes for these two files in a separate commit

File: modules/store-devtools/schematics/ng-add/schema.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      "visible": false
+    },
+    "module": {
+      "type": "string",
+      "default": "app",
+      "description": "Allows specification of the declaring module.",
+      "alias": "m",
+      "subtype": "filepath"
+    },
+    "statePath": {
```
Comment (brandonroberts) [inline modules/store-devtools/schematics/ng-add/schema.json]: The `statePath` property should be removed

File: modules/store-devtools/schematics/ng-add/index.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        module: options.module,
+        path: options.path,
+      });
+    }
+
+    const parsedPath = parseName(options.path, '');
+    options.path = parsedPath.path;
+
+    if ((!options.maxAge && options.maxAge !== 0 ? 25 : options.maxAge) < 0) {
+      options.maxAge = 25;
```
Comment (brandonroberts) [inline modules/store-devtools/schematics/ng-add/index.ts]: This should thrown a `SchematicsException` if the `maxAge` is invalid. The maxAge can be no less than 2 if defined.

File: modules/store-devtools/schematics/ng-add/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import {
+  SchematicTestRunner,
+  UnitTestTree,
+} from '@angular-devkit/schematics/testing';
+import { getFileContent } from '@schematics/angular/utility/test';
+import * as path from 'path';
+import { Schema as RootStoreOptions } from './schema';
```
Comment (brandonroberts) [inline modules/store-devtools/schematics/ng-add/index.spec.ts]: `StoreDevtoolsOptions`

File: modules/store-devtools/schematics/ng-add/index.spec.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  beforeEach(() => {
+    appTree = createWorkspace(schematicRunner, appTree);
+  });
+
+  it('should update package.json', () => {
+    const options = { ...defaultOptions };
+
+    const tree = schematicRunner.runSchematic('ng-add', options, appTree);
+    const packageJson = JSON.parse(getFileContent(tree, '/package.json'));
```
Comment (brandonroberts) [inline modules/store-devtools/schematics/ng-add/index.spec.ts]: Use `tree.readContent('/package.json')` to read the file contents

File: modules/store-devtools/schematics/ng-add/index.spec.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    expect(packageJson.dependencies['@ngrx/store-devtools']).toBeDefined();
+  });
+
+  it('should skip package.json update', () => {
+    const options = { ...defaultOptions, skipPackageJson: true };
+
+    const tree = schematicRunner.runSchematic('ng-add', options, appTree);
+    const packageJson = JSON.parse(getFileContent(tree, '/package.json'));
```
Comment (brandonroberts) [inline modules/store-devtools/schematics/ng-add/index.spec.ts]: Use `tree.readContent('/package.json')` to read the file contents

File: modules/store-devtools/schematics/ng-add/index.ts [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+  findModuleFromOptions,
+  getProjectPath,
+  insertImport,
+  addPackageToPackageJson,
+  platformVersion,
+  parseName,
+} from '@ngrx/store-devtools/schematics-core';
+import { Path, dirname } from '@angular-devkit/core';
+import * as ts from 'typescript';
+import { Schema as RootStoreOptions } from './schema';
```
Comment (brandonroberts) [inline modules/store-devtools/schematics/ng-add/index.ts]: `StoreDevtoolsOptioons`

**Comment by martzcodes** [comment]
@brandonroberts I made your requested changes.  Only tweak being

```
    if (options.maxAge! < 0 || options.maxAge === 1) {
      throw new SchematicsException(`maxAge of ${options.maxAge} is invalid`);
    }
```

changed that since 0 is valid (I'm assuming the schema doesn't support mixed type properties, so 0 would need to be used in lieu of false)

**Comment by brandonroberts** [comment]
One small change. For the error message, `maxAge should be an integer greater than 1`. That way we give them some direction instead of just saying its wrong.

**Comment by martzcodes** [comment]
@brandonroberts updated

**Comment by brandonroberts** [comment]
Great. Thanks!

**Pull request merged**